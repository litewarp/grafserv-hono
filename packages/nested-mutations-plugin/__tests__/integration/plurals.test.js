const { graphql } = require('graphql');
const { withSchema } = require('../helpers');

test(
  'plural when one-to-many, singular in reverse',
  withSchema({
    setup: `
      create table p.parent (
        id serial primary key,
        name text not null
      );
      
      create table p.child (
        id serial primary key,
        parent_id integer,
        name text not null,
        constraint child_parent_fkey foreign key (parent_id)
          references p.parent (id)
      );   
    `,
    test: async ({ schema, pgClient }) => {
      const query = `
        mutation {
          c1: createParent(
            input: {
              parent: {
                name: "test"
                childrenUsingId: {
                  create: [{
                    name: "test child"
                  }]
                }
              }
            }
          ) {
            parent {
              id
            }
          }

          c2: createChild(
            input: {
              child: {
                name: "child"
                parentToParentId: {
                  create: {
                    name: "child's parent"
                  }
                }
              }
            }
          ) {
            child {
              id
            }
          }
        }
      `;
      expect(schema).toMatchSnapshot();

      const result = await graphql(schema, query, null, { pgClient });
      expect(result).not.toHaveProperty('errors');
    },
  }),
);

test(
  'singular when one-to-one',
  withSchema({
    setup: `
      create table p.parent (
        id serial primary key,
        name text not null
      );
      
      create table p.child (
        parent_id serial primary key,
        name text not null,
        constraint child_parent_fkey foreign key (parent_id)
          references p.parent (id)
      );   
    `,
    test: async ({ schema, pgClient }) => {
      const query = `
        mutation {
          c1: createParent(
            input: {
              parent: {
                name: "test"
                childUsingId: {
                  create: {
                    name: "test child"
                  }
                }
              }
            }
          ) {
            parent {
              id
            }
          }

          c2: createChild(
            input: {
              child: {
                name: "child"
                parentToParentId: {
                  create: {
                    name: "child's parent"
                  }
                }
              }
            }
          ) {
            child {
              parentId
            }
          }
        }
      `;
      expect(schema).toMatchSnapshot();

      const result = await graphql(schema, query, null, { pgClient });
      expect(result).not.toHaveProperty('errors');
    },
  }),
);

// https://github.com/mlipscombe/postgraphile-plugin-nested-mutations/issues/7
test(
  '1:1 relationship does not allow multiple nested rows',
  withSchema({
    setup: `
      CREATE TABLE p.post (
        id SERIAL PRIMARY KEY,
        text TEXT NOT NULL
      );
      
      CREATE TABLE p.post_image (
        id SERIAL PRIMARY KEY,
        post_id INTEGER UNIQUE NOT NULL REFERENCES p.post(id) ON DELETE CASCADE,
        url TEXT NOT NULL
      );
    `,
    test: async ({ schema, pgClient }) => {
      const query = `
        mutation {
          c1: createPost(
            input: {
              post: {
                text: "test"
                postImageUsingId: {
                  create: {
                    url: "test child"
                  }
                }
              }
            }
          ) {
            post {
              id
            }
          }

          c2: createPostImage(
            input: {
              postImage: {
                url: "child"
                postToPostId: {
                  create: {
                    text: "child's parent"
                  }
                }
              }
            }
          ) {
            postImage {
              postId
            }
          }
        }
      `;
      expect(schema).toMatchSnapshot();

      const result = await graphql(schema, query, null, { pgClient });
      expect(result).not.toHaveProperty('errors');
    },
  }),
);

test(
  '1:1 relationship mutation fails when multiple operators are specified',
  withSchema({
    setup: `
      CREATE TABLE p.post (
        id SERIAL PRIMARY KEY,
        text TEXT NOT NULL
      );
      
      CREATE TABLE p.post_image (
        id SERIAL PRIMARY KEY,
        post_id INTEGER UNIQUE NOT NULL REFERENCES p.post(id) ON DELETE CASCADE,
        url TEXT NOT NULL
      );
    `,
    test: async ({ schema, pgClient }) => {
      const query = `
        mutation {
          c1: createPost(
            input: {
              post: {
                text: "test"
                postImageUsingId: {
                  create: {
                    url: "test child"
                  }
                  connectById: {
                    id: 1
                  }
                }
              }
            }
          ) {
            post {
              id
            }
          }

          c2: createPostImage(
            input: {
              postImage: {
                url: "child"
                postToPostId: {
                  create: {
                    text: "child's parent"
                  }
                  connectById: {
                    id: 1
                  }
                }
              }
            }
          ) {
            postImage {
              postId
            }
          }
        }
      `;
      expect(schema).toMatchSnapshot();

      const result = await graphql(schema, query, null, { pgClient });
      expect(result).toHaveProperty('errors');
      expect(result.errors[0].message).toMatch(
        /may only create or connect a single row/,
      );
    },
  }),
);
