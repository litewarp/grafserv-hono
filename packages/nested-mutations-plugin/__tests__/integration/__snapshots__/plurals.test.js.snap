// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1:1 relationship does not allow multiple nested rows 1`] = `
GraphQLSchema {
  "__validationErrors": Array [],
  "_directives": Array [
    "@include",
    "@skip",
    "@deprecated",
    "@specifiedBy",
  ],
  "_implementationsMap": Object {
    "Node": Object {
      "interfaces": Array [],
      "objects": Array [
        "Query",
        "Post",
        "PostImage",
      ],
    },
  },
  "_mutationType": "Mutation",
  "_queryType": "Query",
  "_subTypeMap": Object {},
  "_subscriptionType": undefined,
  "_typeMap": Object {
    "Boolean": "Boolean",
    "CreatePostImageInput": "CreatePostImageInput",
    "CreatePostImagePayload": "CreatePostImagePayload",
    "CreatePostInput": "CreatePostInput",
    "CreatePostPayload": "CreatePostPayload",
    "Cursor": "Cursor",
    "DeletePostByIdInput": "DeletePostByIdInput",
    "DeletePostImageByIdInput": "DeletePostImageByIdInput",
    "DeletePostImageByPostIdInput": "DeletePostImageByPostIdInput",
    "DeletePostImageInput": "DeletePostImageInput",
    "DeletePostImagePayload": "DeletePostImagePayload",
    "DeletePostInput": "DeletePostInput",
    "DeletePostPayload": "DeletePostPayload",
    "ID": "ID",
    "Int": "Int",
    "Mutation": "Mutation",
    "Node": "Node",
    "PageInfo": "PageInfo",
    "Post": "Post",
    "PostCondition": "PostCondition",
    "PostImage": "PostImage",
    "PostImageCondition": "PostImageCondition",
    "PostImageInput": "PostImageInput",
    "PostImageNodeIdConnect": "PostImageNodeIdConnect",
    "PostImageNodeIdDelete": "PostImageNodeIdDelete",
    "PostImageOnPostImageForPostImagePostIdFkeyNodeIdUpdate": "PostImageOnPostImageForPostImagePostIdFkeyNodeIdUpdate",
    "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePkeyUpdate": "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePkeyUpdate",
    "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePostIdKeyUpdate": "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePostIdKeyUpdate",
    "PostImagePatch": "PostImagePatch",
    "PostImagePostIdFkeyInput": "PostImagePostIdFkeyInput",
    "PostImagePostIdFkeyInverseInput": "PostImagePostIdFkeyInverseInput",
    "PostImagePostIdFkeyPostCreateInput": "PostImagePostIdFkeyPostCreateInput",
    "PostImagePostIdFkeyPostImageCreateInput": "PostImagePostIdFkeyPostImageCreateInput",
    "PostImagePostImagePkeyConnect": "PostImagePostImagePkeyConnect",
    "PostImagePostImagePkeyDelete": "PostImagePostImagePkeyDelete",
    "PostImagePostImagePostIdKeyConnect": "PostImagePostImagePostIdKeyConnect",
    "PostImagePostImagePostIdKeyDelete": "PostImagePostImagePostIdKeyDelete",
    "PostImagesConnection": "PostImagesConnection",
    "PostImagesEdge": "PostImagesEdge",
    "PostImagesOrderBy": "PostImagesOrderBy",
    "PostInput": "PostInput",
    "PostNodeIdConnect": "PostNodeIdConnect",
    "PostNodeIdDelete": "PostNodeIdDelete",
    "PostOnPostImageForPostImagePostIdFkeyNodeIdUpdate": "PostOnPostImageForPostImagePostIdFkeyNodeIdUpdate",
    "PostOnPostImageForPostImagePostIdFkeyUsingPostPkeyUpdate": "PostOnPostImageForPostImagePostIdFkeyUsingPostPkeyUpdate",
    "PostPatch": "PostPatch",
    "PostPostPkeyConnect": "PostPostPkeyConnect",
    "PostPostPkeyDelete": "PostPostPkeyDelete",
    "PostsConnection": "PostsConnection",
    "PostsEdge": "PostsEdge",
    "PostsOrderBy": "PostsOrderBy",
    "Query": "Query",
    "String": "String",
    "UpdatePostByIdInput": "UpdatePostByIdInput",
    "UpdatePostImageByIdInput": "UpdatePostImageByIdInput",
    "UpdatePostImageByPostIdInput": "UpdatePostImageByPostIdInput",
    "UpdatePostImageInput": "UpdatePostImageInput",
    "UpdatePostImagePayload": "UpdatePostImagePayload",
    "UpdatePostInput": "UpdatePostInput",
    "UpdatePostPayload": "UpdatePostPayload",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
    "updatePostImageOnPostImageForPostImagePostIdFkeyPatch": "updatePostImageOnPostImageForPostImagePostIdFkeyPatch",
    "updatePostOnPostImageForPostImagePostIdFkeyPatch": "updatePostOnPostImageForPostImagePostIdFkeyPatch",
  },
  "astNode": undefined,
  "description": undefined,
  "extensionASTNodes": undefined,
  "extensions": undefined,
}
`;

exports[`1:1 relationship mutation fails when multiple operators are specified 1`] = `
GraphQLSchema {
  "__validationErrors": Array [],
  "_directives": Array [
    "@include",
    "@skip",
    "@deprecated",
    "@specifiedBy",
  ],
  "_implementationsMap": Object {
    "Node": Object {
      "interfaces": Array [],
      "objects": Array [
        "Query",
        "Post",
        "PostImage",
      ],
    },
  },
  "_mutationType": "Mutation",
  "_queryType": "Query",
  "_subTypeMap": Object {},
  "_subscriptionType": undefined,
  "_typeMap": Object {
    "Boolean": "Boolean",
    "CreatePostImageInput": "CreatePostImageInput",
    "CreatePostImagePayload": "CreatePostImagePayload",
    "CreatePostInput": "CreatePostInput",
    "CreatePostPayload": "CreatePostPayload",
    "Cursor": "Cursor",
    "DeletePostByIdInput": "DeletePostByIdInput",
    "DeletePostImageByIdInput": "DeletePostImageByIdInput",
    "DeletePostImageByPostIdInput": "DeletePostImageByPostIdInput",
    "DeletePostImageInput": "DeletePostImageInput",
    "DeletePostImagePayload": "DeletePostImagePayload",
    "DeletePostInput": "DeletePostInput",
    "DeletePostPayload": "DeletePostPayload",
    "ID": "ID",
    "Int": "Int",
    "Mutation": "Mutation",
    "Node": "Node",
    "PageInfo": "PageInfo",
    "Post": "Post",
    "PostCondition": "PostCondition",
    "PostImage": "PostImage",
    "PostImageCondition": "PostImageCondition",
    "PostImageInput": "PostImageInput",
    "PostImageNodeIdConnect": "PostImageNodeIdConnect",
    "PostImageNodeIdDelete": "PostImageNodeIdDelete",
    "PostImageOnPostImageForPostImagePostIdFkeyNodeIdUpdate": "PostImageOnPostImageForPostImagePostIdFkeyNodeIdUpdate",
    "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePkeyUpdate": "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePkeyUpdate",
    "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePostIdKeyUpdate": "PostImageOnPostImageForPostImagePostIdFkeyUsingPostImagePostIdKeyUpdate",
    "PostImagePatch": "PostImagePatch",
    "PostImagePostIdFkeyInput": "PostImagePostIdFkeyInput",
    "PostImagePostIdFkeyInverseInput": "PostImagePostIdFkeyInverseInput",
    "PostImagePostIdFkeyPostCreateInput": "PostImagePostIdFkeyPostCreateInput",
    "PostImagePostIdFkeyPostImageCreateInput": "PostImagePostIdFkeyPostImageCreateInput",
    "PostImagePostImagePkeyConnect": "PostImagePostImagePkeyConnect",
    "PostImagePostImagePkeyDelete": "PostImagePostImagePkeyDelete",
    "PostImagePostImagePostIdKeyConnect": "PostImagePostImagePostIdKeyConnect",
    "PostImagePostImagePostIdKeyDelete": "PostImagePostImagePostIdKeyDelete",
    "PostImagesConnection": "PostImagesConnection",
    "PostImagesEdge": "PostImagesEdge",
    "PostImagesOrderBy": "PostImagesOrderBy",
    "PostInput": "PostInput",
    "PostNodeIdConnect": "PostNodeIdConnect",
    "PostNodeIdDelete": "PostNodeIdDelete",
    "PostOnPostImageForPostImagePostIdFkeyNodeIdUpdate": "PostOnPostImageForPostImagePostIdFkeyNodeIdUpdate",
    "PostOnPostImageForPostImagePostIdFkeyUsingPostPkeyUpdate": "PostOnPostImageForPostImagePostIdFkeyUsingPostPkeyUpdate",
    "PostPatch": "PostPatch",
    "PostPostPkeyConnect": "PostPostPkeyConnect",
    "PostPostPkeyDelete": "PostPostPkeyDelete",
    "PostsConnection": "PostsConnection",
    "PostsEdge": "PostsEdge",
    "PostsOrderBy": "PostsOrderBy",
    "Query": "Query",
    "String": "String",
    "UpdatePostByIdInput": "UpdatePostByIdInput",
    "UpdatePostImageByIdInput": "UpdatePostImageByIdInput",
    "UpdatePostImageByPostIdInput": "UpdatePostImageByPostIdInput",
    "UpdatePostImageInput": "UpdatePostImageInput",
    "UpdatePostImagePayload": "UpdatePostImagePayload",
    "UpdatePostInput": "UpdatePostInput",
    "UpdatePostPayload": "UpdatePostPayload",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
    "updatePostImageOnPostImageForPostImagePostIdFkeyPatch": "updatePostImageOnPostImageForPostImagePostIdFkeyPatch",
    "updatePostOnPostImageForPostImagePostIdFkeyPatch": "updatePostOnPostImageForPostImagePostIdFkeyPatch",
  },
  "astNode": undefined,
  "description": undefined,
  "extensionASTNodes": undefined,
  "extensions": undefined,
}
`;

exports[`plural when one-to-many, singular in reverse 1`] = `
GraphQLSchema {
  "__validationErrors": Array [],
  "_directives": Array [
    "@include",
    "@skip",
    "@deprecated",
    "@specifiedBy",
  ],
  "_implementationsMap": Object {
    "Node": Object {
      "interfaces": Array [],
      "objects": Array [
        "Query",
        "Child",
        "Parent",
      ],
    },
  },
  "_mutationType": "Mutation",
  "_queryType": "Query",
  "_subTypeMap": Object {},
  "_subscriptionType": undefined,
  "_typeMap": Object {
    "Boolean": "Boolean",
    "Child": "Child",
    "ChildChildPkeyConnect": "ChildChildPkeyConnect",
    "ChildChildPkeyDelete": "ChildChildPkeyDelete",
    "ChildCondition": "ChildCondition",
    "ChildInput": "ChildInput",
    "ChildNodeIdConnect": "ChildNodeIdConnect",
    "ChildNodeIdDelete": "ChildNodeIdDelete",
    "ChildOnChildForChildParentFkeyNodeIdUpdate": "ChildOnChildForChildParentFkeyNodeIdUpdate",
    "ChildOnChildForChildParentFkeyUsingChildPkeyUpdate": "ChildOnChildForChildParentFkeyUsingChildPkeyUpdate",
    "ChildParentFkeyChildCreateInput": "ChildParentFkeyChildCreateInput",
    "ChildParentFkeyInput": "ChildParentFkeyInput",
    "ChildParentFkeyInverseInput": "ChildParentFkeyInverseInput",
    "ChildParentFkeyParentCreateInput": "ChildParentFkeyParentCreateInput",
    "ChildPatch": "ChildPatch",
    "ChildrenConnection": "ChildrenConnection",
    "ChildrenEdge": "ChildrenEdge",
    "ChildrenOrderBy": "ChildrenOrderBy",
    "CreateChildInput": "CreateChildInput",
    "CreateChildPayload": "CreateChildPayload",
    "CreateParentInput": "CreateParentInput",
    "CreateParentPayload": "CreateParentPayload",
    "Cursor": "Cursor",
    "DeleteChildByIdInput": "DeleteChildByIdInput",
    "DeleteChildInput": "DeleteChildInput",
    "DeleteChildPayload": "DeleteChildPayload",
    "DeleteParentByIdInput": "DeleteParentByIdInput",
    "DeleteParentInput": "DeleteParentInput",
    "DeleteParentPayload": "DeleteParentPayload",
    "ID": "ID",
    "Int": "Int",
    "Mutation": "Mutation",
    "Node": "Node",
    "PageInfo": "PageInfo",
    "Parent": "Parent",
    "ParentCondition": "ParentCondition",
    "ParentInput": "ParentInput",
    "ParentNodeIdConnect": "ParentNodeIdConnect",
    "ParentNodeIdDelete": "ParentNodeIdDelete",
    "ParentOnChildForChildParentFkeyNodeIdUpdate": "ParentOnChildForChildParentFkeyNodeIdUpdate",
    "ParentOnChildForChildParentFkeyUsingParentPkeyUpdate": "ParentOnChildForChildParentFkeyUsingParentPkeyUpdate",
    "ParentParentPkeyConnect": "ParentParentPkeyConnect",
    "ParentParentPkeyDelete": "ParentParentPkeyDelete",
    "ParentPatch": "ParentPatch",
    "ParentsConnection": "ParentsConnection",
    "ParentsEdge": "ParentsEdge",
    "ParentsOrderBy": "ParentsOrderBy",
    "Query": "Query",
    "String": "String",
    "UpdateChildByIdInput": "UpdateChildByIdInput",
    "UpdateChildInput": "UpdateChildInput",
    "UpdateChildPayload": "UpdateChildPayload",
    "UpdateParentByIdInput": "UpdateParentByIdInput",
    "UpdateParentInput": "UpdateParentInput",
    "UpdateParentPayload": "UpdateParentPayload",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
    "updateChildOnChildForChildParentFkeyPatch": "updateChildOnChildForChildParentFkeyPatch",
    "updateParentOnChildForChildParentFkeyPatch": "updateParentOnChildForChildParentFkeyPatch",
  },
  "astNode": undefined,
  "description": undefined,
  "extensionASTNodes": undefined,
  "extensions": undefined,
}
`;

exports[`singular when one-to-one 1`] = `
GraphQLSchema {
  "__validationErrors": Array [],
  "_directives": Array [
    "@include",
    "@skip",
    "@deprecated",
    "@specifiedBy",
  ],
  "_implementationsMap": Object {
    "Node": Object {
      "interfaces": Array [],
      "objects": Array [
        "Query",
        "Child",
        "Parent",
      ],
    },
  },
  "_mutationType": "Mutation",
  "_queryType": "Query",
  "_subTypeMap": Object {},
  "_subscriptionType": undefined,
  "_typeMap": Object {
    "Boolean": "Boolean",
    "Child": "Child",
    "ChildChildPkeyConnect": "ChildChildPkeyConnect",
    "ChildChildPkeyDelete": "ChildChildPkeyDelete",
    "ChildCondition": "ChildCondition",
    "ChildInput": "ChildInput",
    "ChildNodeIdConnect": "ChildNodeIdConnect",
    "ChildNodeIdDelete": "ChildNodeIdDelete",
    "ChildOnChildForChildParentFkeyNodeIdUpdate": "ChildOnChildForChildParentFkeyNodeIdUpdate",
    "ChildOnChildForChildParentFkeyUsingChildPkeyUpdate": "ChildOnChildForChildParentFkeyUsingChildPkeyUpdate",
    "ChildParentFkeyChildCreateInput": "ChildParentFkeyChildCreateInput",
    "ChildParentFkeyInput": "ChildParentFkeyInput",
    "ChildParentFkeyInverseInput": "ChildParentFkeyInverseInput",
    "ChildParentFkeyParentCreateInput": "ChildParentFkeyParentCreateInput",
    "ChildPatch": "ChildPatch",
    "ChildrenConnection": "ChildrenConnection",
    "ChildrenEdge": "ChildrenEdge",
    "ChildrenOrderBy": "ChildrenOrderBy",
    "CreateChildInput": "CreateChildInput",
    "CreateChildPayload": "CreateChildPayload",
    "CreateParentInput": "CreateParentInput",
    "CreateParentPayload": "CreateParentPayload",
    "Cursor": "Cursor",
    "DeleteChildByParentIdInput": "DeleteChildByParentIdInput",
    "DeleteChildInput": "DeleteChildInput",
    "DeleteChildPayload": "DeleteChildPayload",
    "DeleteParentByIdInput": "DeleteParentByIdInput",
    "DeleteParentInput": "DeleteParentInput",
    "DeleteParentPayload": "DeleteParentPayload",
    "ID": "ID",
    "Int": "Int",
    "Mutation": "Mutation",
    "Node": "Node",
    "PageInfo": "PageInfo",
    "Parent": "Parent",
    "ParentCondition": "ParentCondition",
    "ParentInput": "ParentInput",
    "ParentNodeIdConnect": "ParentNodeIdConnect",
    "ParentNodeIdDelete": "ParentNodeIdDelete",
    "ParentOnChildForChildParentFkeyNodeIdUpdate": "ParentOnChildForChildParentFkeyNodeIdUpdate",
    "ParentOnChildForChildParentFkeyUsingParentPkeyUpdate": "ParentOnChildForChildParentFkeyUsingParentPkeyUpdate",
    "ParentParentPkeyConnect": "ParentParentPkeyConnect",
    "ParentParentPkeyDelete": "ParentParentPkeyDelete",
    "ParentPatch": "ParentPatch",
    "ParentsConnection": "ParentsConnection",
    "ParentsEdge": "ParentsEdge",
    "ParentsOrderBy": "ParentsOrderBy",
    "Query": "Query",
    "String": "String",
    "UpdateChildByParentIdInput": "UpdateChildByParentIdInput",
    "UpdateChildInput": "UpdateChildInput",
    "UpdateChildPayload": "UpdateChildPayload",
    "UpdateParentByIdInput": "UpdateParentByIdInput",
    "UpdateParentInput": "UpdateParentInput",
    "UpdateParentPayload": "UpdateParentPayload",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
    "updateChildOnChildForChildParentFkeyPatch": "updateChildOnChildForChildParentFkeyPatch",
    "updateParentOnChildForChildParentFkeyPatch": "updateParentOnChildForChildParentFkeyPatch",
  },
  "astNode": undefined,
  "description": undefined,
  "extensionASTNodes": undefined,
  "extensions": undefined,
}
`;
